components/apps_form/apps_form_field/select_channel_option.tsx

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {components, OptionProps} from 'react-select';
+
+import {AppSelectOption} from '@mattermost/types/apps';
+
+const {Option} = components;
+export const SelectChannelOption = (props: OptionProps<AppSelectOption>) => {
```

Comment:
If we make this a `function`, we can export as default in one line

```suggestion
export default function SelectChannelOption(props: OptionProps<AppSelectOption>) {
```